wger/exercises/tests/test_exercise.py

```diff
@@ -422,6 +422,38 @@ def search_exercise(self, fail=True):
         result = json.loads(response.content.decode("utf8"))
         self.assertEqual(len(result["suggestions"]), 0)
 
+    def filter_exercise(self, fail=True):
+        """
+        Helper function to test getting all exercise info
+        """
+
+        # 1 hit, "Very cool exercise"
+        response = self.client.get(
+            reverse("exercise-info"), {"term": "cool"}
+        )
+        self.assertEqual(response.status_code, 200)
+        result = json.loads(response.content.decode("utf8"))
+        self.assertEqual(len(result), 1)
+        self.assertEqual(
+            result["info"][0]["value"], "Very cool exercise"
+        )
+        self.assertEqual(result["info"][0]["data"]["id"], 2)
+        self.assertEqual(
+            result["info"][0]["data"]["category"], "Another category"
+        )
+        self.assertEqual(result["info"][0]["data"]["image"], None)
```

Comment:
These guys used to test all the test cases in one function but it's, not a good test practice as it affects many tests at once and sometimes when they accumulate in a function, it gets hard to tell which one causes the break in case the tests break.

Reply:
@NicholusMuwonge, thanks for the feedback however I have two test cases and one function. the reason I put all the tests in that function was to cover the two cases when a user is logged in. Instead of writing two test cases for each scenario when a user is logged in and not logged in which would lead to redundancy

Reply:
Oh, okay sorry @frankopkusianwar, my bad